### PR TITLE
#4378 against master

### PR DESCRIPTION
**Pull request by ottenhoff**
untested code. please note the FIXMEs in the code. 

This is the #4378 PR except against master

CC: @tawillis

### CODE REVIEW

File: edu-services/gradebook-service/impl/src/java/org/sakaiproject/component/gradebook/GradebookPermissionServiceImpl.java [hunk 1]
```diff
@@ -1178,6 +1179,59 @@ public List getViewableStudentsForUser(String gradebookUid, String userId, List
...
+	 * @param role user Role
+	 * @return list of {@link org.sakaiproject.service.gradebook.shared.PermissionDefinition PermissionDefinitions} or empty list if none
+	 */
+	public List<PermissionDefinition> getRealmsPermissionsForUser(String userUuid,String siteId, Role role){
+
+		List<PermissionDefinition> permissions = new ArrayList<PermissionDefinition>();
+
+		if( this.getGradebookService().isUserAllowedToGrade(siteId,userUuid)){
+			//FIXME:giving them view course grade (this needs to be reviewed!!), 
+			//it appears in GB classic, User can view course grades if they have the ability to grade in realms
```
Comment (ottenhoff) [inline edu-services/gradebook-service/impl/src/java/org/sakaiproject/component/gradebook/GradebookPermissionServiceImpl.java]: I don't understand this FIXME. Steve?

Comment (tawillis) [inline edu-services/gradebook-service/impl/src/java/org/sakaiproject/component/gradebook/GradebookPermissionServiceImpl.java]: I think the issue with the FIXME is that GBNG has a view course grades permission that doesn't exist in realms. GB classic allowed a user to view course grades if they have grading permissions. I made a comment that it needs to be reviewed as I am not sure what the direction is for GBNG, why the view course grade permission was added. Without it, TA's cannot view course grades unless they are given the privilege through GBNG permissions. The proper solution may be to add the permission to realms templates.

Steve, many thanks for merging this to master!

Comment (steveswinsburg) [inline edu-services/gradebook-service/impl/src/java/org/sakaiproject/component/gradebook/GradebookPermissionServiceImpl.java]: Yeah viewing course grades is a new permission in GBNG that can be given to TA's. @kyleblythe I cannot remember the rationale for this, do we have doco? Should have created feature files.

**Comment by ern** [comment]
The title on this PR could be better ;)

**Pull request closed without merging**